src/network/networkpacket.cpp

```diff
@@ -160,9 +162,16 @@ NetworkPacket& NetworkPacket::operator>>(std::wstring& dst)
 	checkReadOffset(m_read_offset, strLen * 2);
 
 	dst.reserve(strLen);
-	for(u16 i=0; i<strLen; i++) {
-		wchar_t c16 = readU16(&m_data[m_read_offset]);
-		dst.append(&c16, 1);
+	for (u16 i = 0; i < strLen; i++) {
+		wchar_t c = readU16(&m_data[m_read_offset]);
+		if (NEED_SURROGATE_CODING && c >= 0xD800 && c < 0xDC00 && i+1 < strLen) {
+			i++;
+			m_read_offset += sizeof(u16);
+
+			wchar_t c2 = readU16(&m_data[m_read_offset]);
+			c = 0x10000 + ((c & 0x3ff) << 10) | (c2 & 0x3ff);
```

Comment:
Does it work as intended?
```
warning: suggest parentheses around arithmetic in operand of ‘|’ [-Wparentheses]
  172 |    c = 0x10000 + ((c & 0x3ff) << 10) | (c2 & 0x3ff);
```

Reply:
don’t remember the precedence but it should work either way IIUC

Reply:
This supposed to be `0x10000 + ( ((c & 0x3ff) << 10) | (c2 & 0x3ff) );`, will change.